bluepyopt/ephys/morphologies.py

```diff
@@ -172,43 +180,60 @@ def set_nseg(self, icell):
             section.nseg = 1 + 2 * int(section.L / self.nseg_frequency)
 
     @staticmethod
-    def replace_axon(sim=None, icell=None):
+    def replace_axon(sim=None, icell=None, axon_stump_length=60, n_sections=2):
         """Replace axon"""
 
         nsec = len([sec for sec in icell.axonal])
 
+        print(f"In replace_axon {nsec = }")
+
         if nsec == 0:
-            ais_diams = [1, 1]
+            ais_diams = [1] * n_sections
         elif nsec == 1:
-            ais_diams = [icell.axon[0].diam, icell.axon[0].diam]
+            sec_x = [min(1, x/n_sections * axon_stump_length/icell.axon[0].L)
+                     for x in range(0, n_sections)]
+            ais_diams = [icell.axon[0](sx).diam for sx in sec_x]
         else:
-            ais_diams = [icell.axon[0].diam, icell.axon[0].diam]
+            ais_diams = [icell.axon[0].diam] * n_sections
             # Define origin of distance function
             sim.neuron.h.distance(0, 0.5, sec=icell.soma[0])
 
+            end_point_dist = np.arange(1, n_sections+1) \
+                * axon_stump_length / n_sections
+
+            idx = 1
+
             for section in icell.axonal:
-                # If distance to soma is larger than 60, store diameter
-                if sim.neuron.h.distance(1, 0.5, sec=section) > 60:
-                    ais_diams[1] = section.diam
+                if idx >= len(ais_diams):
                     break
+                if sim.neuron.h.distance(1, 0.5, sec=section)\
```

Comment:
I think that here also you are changing the default behavior

Reply:
I believe the default behaviour should be retained now. Let me know if there is anything else?